dtbase/models/arima.py

```
@@ -0,0 +1,550 @@
+from dtbase.models.utils.dataprocessor.prepare_data import prepare_data
+from dtbase.services.base import BaseModel
+
+# Read default params from config file
+_config_file_path = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "config_arima.ini"
+)
+_defaults = read_config(_config_file_path, "arima")
+
+# set up logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+field_styles = coloredlogs.DEFAULT_FIELD_STYLES
+
+# change the default levelname color from black to yellow
+field_styles["levelname"]["color"] = "yellow"
+coloredlogs.ColoredFormatter(field_styles=field_styles)
+coloredlogs.install(level="INFO")
+
+
+class ConfigArima(pydantic.BaseModel):
+    hours_forecast: int = pydantic.Field(
+        default=_defaults["hours_forecast"], validate_default=True
+    )
+    arima_order: tuple[int, int, int] = pydantic.Field(
+        default=_defaults["arima_order"], validate_default=True
+    )
+    seasonal_order: tuple[int, int, int, int] = pydantic.Field(
+        default=_defaults["seasonal_order"], validate_default=True
+    )
+    trend: str | list[int] | None = pydantic.Field(
+        default=_defaults["trend"], validate_default=True
+    )
+    alpha: float = pydantic.Field(default=_defaults["alpha"], validate_default=True)
+    perform_cv: bool = pydantic.Field(
+        default=_defaults["perform_cv"], validate_default=True
+    )
+    cv_refit: bool = pydantic.Field(
+        default=_defaults["cv_refit"], validate_default=True
+    )
+
+
+class ArimaModel(BaseModel):
+    def __init__(self, config: Optional[dict] = None) -> None:
+        super().__init__()
+        self.config = config
+
+    @staticmethod
+    def _report_errors_for_data(data: pd.Series) -> None:
+        """
+        Report errors if the input data is not a pandas Series indexed by timestamp.
+        """
+        if not isinstance(data.index, pd.DatetimeIndex):
+            logger.error(
```

Rather than logging an error and raising a `ValueError`, could we just raise `raise ValueError("blahblah explanation")`?

---

Yup, cleaner